### DISCUSSION TITLE
Open `date` yaml to different type of dates

### DISCUSSION THREAD

**Discussion by TheHanziChauvanist** [quarto-dev/quarto-cli#9368] [Q&A] [answered]
### Description

I want the `date` function in yaml to be open to different formats, for example:

+ The hijri and the persian calendar
+ date ranges e.g. 1945-1965
+ 1939 without the need to make it 1939-01-01

There are probably many more that i didn't mention.

**Comment by mcanouil** [comment]
How `date-format` does not address what you want?
See <https://quarto.org/docs/reference/dates.html#date-formatting>.

**Reply by TheHanziChauvanist** [reply]
Yes, I know. However the books that I'm digitizing into Quarto format don't have concrete dates, they have only years to go of off. And it feels wrong to put in 1900-01-01 into the file. 

Also one of the books is a duology with different dates, so I want to put in a date range (first book: 1949, second book: 1955), therefore (1949-1955).

**Reply by mcanouil** [reply, marked as answer]
Sorry, but why do you want to set `1900-01-01` if you only want the year?
Change the format.

````qmd
---
title: "Quarto Playground"
format: html
date: "2024"
date-format: "YYYY"
---

This is a playground for Quarto.
````

Dates are handled by Date.js. Date ranges are not dates, but you can still do them:

````qmd
---
title: "Quarto Playground"
format: html
date: "2024"
date-format: "[2010-]YYYY"
---

This is a playground for Quarto.
````

**Reply by TheHanziChauvanist** [reply]
I know I would never write in non-Gregorian dates, but how can I do that in Quarto, or any other non standard dates? Like Hijri calendars, Persian, Thai, Nepalese, Chinese and Buddhist?

**Reply by TheHanziChauvanist** [reply]
And also, is there plans to extend the functionaility of the `date` YAML or no?

**Reply by mcanouil** [reply]
As said, date is handled by Date.js.

I have no idea what those other dates you are talking about looks like.
Could you actually give concrete examples?
